src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            first = new Node<T>(null, value, null);
+            last = first;
+        } else {
+            last.next = new Node<T>(last, value, null);
+            last = last.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + " out of bounds");
+        }
```

Comment:
You already have a `checkIndex` method, no need to duplicate this code. Try to pass `size` and `size + 1` in different cases as parameter to this method.